Source/Diagnostics/ReducedDiags/ColliderRelevant.cpp

```diff
@@ -0,0 +1,559 @@
+
+    const auto get_idx = [&](const std::string& name){
+        return m_headers_indices.at(name).idx;
+    };
+
+    std::array<std::unique_ptr<amrex::MultiFab>, 2> num_dens;
+
+    // loop over species
+    for (int i_s = 0; i_s < 2; ++i_s)
+    {
+        // get WarpXParticleContainer class object
+        WarpXParticleContainer& myspc = mypc.GetParticleContainerFromName(m_beam_name[i_s]);
+        // get charge
+        amrex::ParticleReal const q = myspc.getCharge();
+
+        using PType = typename WarpXParticleContainer::SuperParticleType;
+
+        num_dens[i_s] = myspc.GetChargeDensity(0);
+        num_dens[i_s]->mult(1./q);
+
+        // wtot
+        amrex::Real wtot = ReduceSum( myspc,
+            [=] AMREX_GPU_HOST_DEVICE (const PType& p)
+            {
+                return p.rdata(PIdx::w);
+            });
+        amrex::ParallelDescriptor::ReduceRealSum(wtot, amrex::ParallelDescriptor::IOProcessorNumber());
+
+#if defined(WARPX_DIM_XZ)
+        // thetax_min, thetax_max
+        amrex::Real thetax_min = 0.0_rt;
+        amrex::Real thetax_max = 0.0_rt;
+
+        amrex::ReduceOps<ReduceOpMin, ReduceOpMax> reduce_ops_minmax;
+        auto r_minmax = amrex::ParticleReduce<amrex::ReduceData<Real, Real>>(
+            myspc,
+            [=] AMREX_GPU_DEVICE(const PType& p) noexcept -> amrex::GpuTuple<Real, Real>
+            {
+                const amrex::Real ux = p.rdata(PIdx::ux);
+                const amrex::Real uz = p.rdata(PIdx::uz);
+                const amrex::Real thetax = std::atan2(ux, uz);
+                return {thetax, thetax};
+            },
+            reduce_ops_minmax);
+        thetax_min = amrex::get<0>(r_minmax);
+        thetax_max = amrex::get<1>(r_minmax);
+        amrex::ParallelDescriptor::ReduceRealMin(thetax_min, amrex::ParallelDescriptor::IOProcessorNumber());
+        amrex::ParallelDescriptor::ReduceRealMax(thetax_max, amrex::ParallelDescriptor::IOProcessorNumber());
+
+        // x_ave, thetax_ave
+        amrex::Real x_ave = 0.0_rt;
+        amrex::Real thetax_ave = 0.0_rt;
+
+        amrex::ReduceOps<ReduceOpSum, ReduceOpSum> reduce_ops_ave;
+        auto r_ave = amrex::ParticleReduce<amrex::ReduceData<Real, Real>>(
+            myspc,
+            [=] AMREX_GPU_DEVICE(const PType& p) noexcept -> amrex::GpuTuple<Real, Real>
+            {
+                const amrex::Real w  = p.rdata(PIdx::w);
+                const amrex::Real x = p.pos(0);
+                const amrex::Real ux = p.rdata(PIdx::ux);
+                const amrex::Real uz = p.rdata(PIdx::uz);
+                const amrex::Real thetax = std::atan2(ux, uz);
+                return {w*x, w*thetax};
+            },
+            reduce_ops_ave);
+        x_ave = amrex::get<0>(r_ave);
+        thetax_ave = amrex::get<1>(r_ave);
+        amrex::ParallelDescriptor::ReduceRealSum({x_ave, thetax_ave}, amrex::ParallelDescriptor::IOProcessorNumber());
+        x_ave = x_ave / wtot;
+        thetax_ave = thetax_ave / wtot;
```

Comment:
What if wtot is zero? Since the mpi_reduce above only reduces to the I/O process, if one of the non-I/O processes has no particles, wtot will be zero on that process. So maybe do this only `if (amrex::ParallelDescriptror::IOProcessor())`.